File: JDK11/opentraceability/src/main/java/opentraceability/models/events/eventsourcetype.java
package opentraceability.models.events;

import java.util.Arrays;
import java.util.List;

import opentraceability.utility.CBVAttribute;

public enum EventSourceType {
    Unknown(0),
    Owner(1,
    		"owning_party",
    		"https://ref.gs1.org/cbv/SDT-owning_party",
    		"urn:epcglobal:cbv:sdt:owning_party"),
    Possessor(2,
    		"possessing_party",
    		"https://ref.gs1.org/cbv/SDT-possessing_party",
    		"urn:epcglobal:cbv:sdt:possessing_party"),
    Location(3,
    		"location",
    		"https://ref.gs1.org/cbv/SDT-location",
    		"urn:epcglobal:cbv:sdt:location");

    private final int value;

    private final List<String> attributes;

    EventSourceType(int value, String...attributes) {
        this.value = value;
        this.attributes = Arrays.asList(attributes);
    }

    public int getValue() {
        return value;
    }

    public List<String> getCBVAttributes() {
        return attributes;
    }
}

File: Java/OpenTraceability/src/main/opentraceability/models/events/eventsourcetype.kt
package models.events

import utility.CBVAttribute

enum class EventSourceType(val value: Int) {
    Unknown(0),

    @CBVAttribute("owning_party")
    @CBVAttribute("https://ref.gs1.org/cbv/SDT-owning_party")
    @CBVAttribute("urn:epcglobal:cbv:sdt:owning_party")
    Owner(1),

    @CBVAttribute("possessing_party")
    @CBVAttribute("https://ref.gs1.org/cbv/SDT-possessing_party")
    @CBVAttribute("urn:epcglobal:cbv:sdt:possessing_party")
    Possessor(2),

    @CBVAttribute("location")
    @CBVAttribute("https://ref.gs1.org/cbv/SDT-location")
    @CBVAttribute("urn:epcglobal:cbv:sdt:location")
    Location(3),
}
